robber/expect.py

```diff
@@ -32,14 +33,14 @@ def remove_custom_message(cls):
         cls.message = None
 
     @classmethod
-    def register(cls, name, klass):
+    def register(cls, name, klass, is_negative=False):
         cls.matchers[name] = klass
 
         def method(self, other=None, *args):
-            return klass(self.object, other, *args).fail_with(self.message).match()
+            fact = is_negative is not self.is_negative
```

Comment:
Well, I agree with you that it's (really) magical. But only the implementation is complicated, and we only have to do it once. This way of registration is really easy to use, DRY and saves time. So in my opinion, the trade-off is acceptable.